frontend/src/components/S3ArtifactLink.tsx

```diff
@@ -0,0 +1,31 @@
+import * as React from 'react';
+import { StoragePath, StorageService } from '../lib/WorkflowParser';
+import { S3Artifact } from '../../third_party/argo-ui/argo_template';
+
+const artifactApiUri = ({ source, bucket, key }: StoragePath) =>
```

Comment:
It looks like this works with minio backed by gcs/local disk too. Is my understanding correct?

If that's the case, shall we rename this component to `MinioArtifactLink`?

Reply:
yup. Have renamed to MinioArtifactLink.